src/iuse.cpp

```diff
@@ -7942,10 +7937,11 @@ int iuse::foodperson( player *p, item *it, bool t, const tripoint &pos )
 int iuse::radiocar( player *p, item *it, bool, const tripoint & )
 {
     int choice = -1;
-    auto bomb_it = std::find_if( it->contents.begin(), it->contents.end(), []( const item & c ) {
+    std::list<item> all_items{ it->contents.all_items() };
```

Comment:
This list contains copies of all the items within the pockets. `bomb_it` contains a reference to one of those copied items.

Later you call `it->contents.remove_item( *bomb_it )`, but `remove_item` does removal based on the item pointer (see `item_pocket::remove_item`) , which is now a pointer to the copy stored in the `all_items` list. It won't remove anything.

This problem might exist at other places that use the result of `all_items`.